json-logic/schema.ts

```diff
@@ -0,0 +1,146 @@
+// LogicExpression refers to the JSONLogic standard.
+import { LogicExpression } from "./logic";
+
+/*
+* This module defines a schema for questionaire and result calulation logic.
+*/
+
+/**
+ * Type of a question, essentially defines the result type.
+ */
+export enum QuestionType {
+	Select = "Select",
+	Multiselect = "Multiselect",
+	Number = "Number",
+	Boolean = "Boolean",
+	Date = "Date",
+	Text = "Text"
+}
+
+/**
+ * Option for multi-select questions.
+ */
+export interface IOption {
+	/**
+	 * Human-Readable answer, can be localized.
+	 */
+	text: string,
+	/**
+	 * Value used for evaluating logic expressions.
+	 */
+	value: string;
+}
+
+/**
+ * Represents a question.
+ */
+export interface IQuestion {
+	/**
+	 * Unique id for referring this question in logic expressions.
+	 */
+	id: string
+	/**
+	 * Type of the question.
+	 */
+	type: QuestionType,
+	/**
+	 * Human-readable question text, can be localized.
+	 */
+	text: string,
+	/**
+	 * Answer options for Select/Multiselect questions.
+	 */
+	options?: IOption[],
+	/**
+	 * Boolean indicating whether the question is optional or not.
+	 */
+	optional?: boolean
+	/**
+	 * Logic expression to decide whether the question should be displayed or not.
+	 */
+	skipIf?: LogicExpression
+}
+
+/**
+ * Represents a variable which is computed from the given answers or other variables.
+ */
+export interface IVariable {
```

Comment:
I assume that question answers are still of type `IVariable` because `IQuestionnaire.variables` is an array of this type. In this case, `value` will be the value of the answer. This is a subtype of `LogicConstant` but maybe we should be more explicit to show that also question answers are `IVariable`.

Reply:
Maybe we can add an attribute `source` which is the question id if it is an answer to a question and `computed` if it is defined by a logical expression.

Reply:
I think that's more sepcific to the actual implementation (e.g. `questionaire.ts`), not to the definition format used to communicate the questionaire.

I'm happy to use the proposed approach in the reference implementation, but I don't think it should be part of our questionnaire format.